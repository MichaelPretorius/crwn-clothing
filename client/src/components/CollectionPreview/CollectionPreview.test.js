import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './CollectionPreview';

describe('CollectionPreview', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const mockRouteName = 'hats';

  beforeEach(() => {
    mockMatch = {
      path: '/shop'
    };

    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: 'hats',
      items: []
    };

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  test('should render CollectionPreview', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  
  test('should call history.push with right string when TitleContainer clicked', () => {
    wrapper.find('TitleContainer').simulate('click');
    expect(mockHistory.push).toBeCalledWith(`${mockMatch.path}/${mockRouteName}`);
  });
})
