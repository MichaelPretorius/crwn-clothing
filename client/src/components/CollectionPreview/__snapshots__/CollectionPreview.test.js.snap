// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview should render CollectionPreview 1`] = `
"<styled.div>
  <TitleContainer onClick={[Function: onClick]}>
    HATS
  </TitleContainer>
  <styled.div />
</styled.div>"
`;
